### PR TITLE
Feature/template files

### PR DESCRIPTION
**Pull request by netdesignr**
Hi @domingobishop,

Can you please review this pull request?

Thank you!

Mihai

### CODE REVIEW

File: css/sass/main.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
+.page-template-page-section-landing {
```
Comment (domjngo) [inline css/sass/main.scss]: Remove if not in use

File: css/sass/main.scss [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  .secondary-content {
+    background-color: #ffffff;
+    padding: 20px;
+    .content-box {
+
+    }
+  }
+}
+p {
```
Comment (domjngo) [inline css/sass/main.scss]: When using rules to override global rules, please use a new class to target these elements.
eg .page-template-class p { margin-bottom:0; }

File: page-research-repository-landing.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    <div id="primary" class="content-area">
+        <div class="container">
+            <div class="row">
+                <main id="main" class="col-xs-12 col-sm-8 col-md-8" role="main">
+                    <?php $feat_image = wp_get_attachment_url(get_post_thumbnail_id($post->ID)); ?>
+                    <?php if (have_posts()) : while (have_posts()) : the_post(); ?>
+                        <article id="post-<?php the_ID(); ?>" <?php post_class(); ?>>
+                            <div class="entry-header img-container"
+                                 style="background-image: url('<?php echo preg_replace('/https?:\/\/research.(live|dev|test)lb\.nationalarchives\.gov\.uk\//', '/', $feat_image); ?>')">
```
Comment (domjngo) [inline page-research-repository-landing.php]: The preg_replace should be inside the function

Comment (domjngo) [reply page-research-repository-landing.php]: Actually, this should return a relative path. Try using make_path_relative function.

File: page-research-repository-details.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                        <article id="post-<?php the_ID(); ?>" <?php post_class(); ?>>
+                            <div class="entry-header">
+                                <h1><?php the_title(); ?></h1>
+                            </div>
+                            <div class="entry-content clearfix">
+                                <span class="entry-meta">
+                                <?php display_authors(); ?>
+                                </span>
+                                <br/>
+                                <span class="entry-meta">
```
Comment (domjngo) [inline page-research-repository-details.php]: Is there a reason why this content (.entry-meta) is a smaller font size?

File: page-research-repository-landing.php (hunk 3 above)
Comment (domjngo) [reply page-research-repository-landing.php]: Background image needs CSS to position and size.

File: page-research-repository-landing.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                            <div class="entry-content clearfix">
+                                <?php the_content(); ?>
+                            </div>
+                        </article>
+                    <?php endwhile;
+                    else: ?>
+                        <p>Sorry, no content</p>
+                    <?php endif;
+                    wp_reset_query(); ?>
+                    <div class="col-xs-12 col-md-12 secondary-content">
```
Comment (domjngo) [inline page-research-repository-landing.php]: col-xs-12 col-md-12 is not needed here

File: page-research-repository-landing.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                                <a href="<?php the_permalink(); ?>" title="<?php the_title(); ?>">
+                                    <h3>
+                                        <?php the_title(); ?>
+                                    </h3>
+                                </a>
+                                <span class="entry-meta">
+                                    <?php display_authors(); ?>
+                                </span>
+                                <p>
+                                    <?php the_excerpt(); ?>
```
Comment (domjngo) [inline page-research-repository-landing.php]: If the excerpt is too long it's displaying [...]

File: inc/functions/functions-templates.php [hunk 7]
```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Display Authors
+ */
+function display_authors()
```
Comment (domjngo) [inline inc/functions/functions-templates.php]: If data doesn't exist should this still return something?

File: inc/functions/functions-templates.php [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            get_sidebar(get_post_meta($post->ID, 'sidebar', true));
+        }
+    }
+    return false;
+}
+
+/**
+ * @return bool
+ */
+function display_date_of_publication()
```
Comment (domjngo) [inline inc/functions/functions-templates.php]: If data doesn't exist should this still return something?

File: inc/functions/functions-templates.php [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+    if (function_exists('get_post_meta')) {
+        echo date('d F Y', strtotime(get_post_meta($post->ID, 'date_published', true)));
+    }
+    return false;
+}
+
+/**
+ * @return bool
+ */
+function display_keywords_taxonomy()
```
Comment (domjngo) [inline inc/functions/functions-templates.php]: If data doesn't exist should this still return something?

File: inc/functions/functions-templates.php [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        }
+    }
+
+    return false;
+}
+
+/**
+ * @return bool
+ */
+function display_published_by()
```
Comment (domjngo) [inline inc/functions/functions-templates.php]: If data doesn't exist should this still return something?

File: inc/functions/functions-templates.php [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+        }
+    }
+
+    return false;
+}
+
+/**
+ * @return bool
+ */
+function display_file_url_file_size()
```
Comment (domjngo) [inline inc/functions/functions-templates.php]: If data doesn't exist should this still return something?

**Review by domjngo — CHANGES_REQUESTED** [review]
Please look at the comments

File: page-research-repository-landing.php (hunk 5 above)
Comment (punalsc) [reply page-research-repository-landing.php]: Done.

File: page-research-repository-landing.php (hunk 3 above)
Comment (punalsc) [reply page-research-repository-landing.php]: CSS background position and size DONE.

File: css/sass/main.scss (hunk 2 above)
Comment (punalsc) [reply css/sass/main.scss]: Done.

File: page-research-repository-landing.php (hunk 3 above)
Comment (punalsc) [reply page-research-repository-landing.php]: Now using make_path_relative().

File: page-research-repository-landing.php (hunk 6 above)
Comment (domjngo) [reply page-research-repository-landing.php]: To be reviewed at tna-base level

File: inc/functions/functions-templates.php (hunk 7 above)
Comment (domjngo) [reply inc/functions/functions-templates.php]: Fixed

File: inc/functions/functions-templates.php (hunk 8 above)
Comment (domjngo) [reply inc/functions/functions-templates.php]: Fixed

File: inc/functions/functions-templates.php (hunk 9 above)
Comment (domjngo) [reply inc/functions/functions-templates.php]: Fixed

File: inc/functions/functions-templates.php (hunk 10 above)
Comment (domjngo) [reply inc/functions/functions-templates.php]: Fixed

**Review by domjngo — APPROVED** [review]

**Pull request merged**